sabnzbd/constants.py

```diff
@@ -40,6 +40,7 @@
 # Leave some space for "_UNPACK_" which we append during post-proc
 # Or, when extra ".1", ".2" etc. are added for identically named jobs
 DEF_FOLDER_MAX = 256 - 10
+DEF_FILE_MAX = 256 - 10
```

Comment:
Why subtract 10 for the filename? Any risk of lengthy tags like those applies to dirs when unpacking?

And IIRC the actual filename length limit on mainstream filesystems is 255.

Reply:
Just trying to stay on the safe side, without too much thinking because of this edge case. I would consider 255 too risky and/or too much work to think about and to test. But ... I can let the unittest take care of the testing and let it write the actual file to see if it works.

... playing with that

>>> outF = open("A"*255, "w")
>>> 
>>> outF = open("A"*256, "w")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
OSError: [Errno 36] File name too long: 'AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA'
>>> 

... OK, good, but then I discovered something bad:

```
>>> name = "你"*86
>>> len(name)
86
>>> outF = open(name, "w")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
OSError: [Errno 36] File name too long: '你你你你你你你你你你你你你你你你你你你你你你你你你你你你你你你你你你你你你你你你你你你你你你你你你你你你你你你你你你你你你你你你你你你你你你你你你你你你你你你你你你你你你你'
```

Ouch: 86 UTF-8 is too much! Which is logical if you look at the real bytes:

```
>>> str.encode(name)
b'\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0\xe4\xbd\xa0'
>>> len(str.encode(name))
258
```
So > 255.

... I have some work to do in the code. Thanks!

Reply:
Nice find on the encoding.

For the record: I didn't check whether sab appends a `.1 .2 ... [.n]` style suffix as a workaround for duplicate filenames. If it does, some margin is obviously needed, although reserving 10 chars (i.e. room for 10<sup>9</sup> duplicates of any single filename) just for that seems excessive.

Reply:
This works:
```
>>> name = "你"*86
>>> len(name.encode("utf8"))
258
```

Reply:
I've set it to 5.